### PR TITLE
TMDM-13379 : [TQL -> MQL] Foreign key Infos in Contains and Equals expressions

### PR DESCRIPTION
**Pull request by YohannTalend**
https://jira.talendforge.org/browse/TMDM-13379

**What is the current behavior?** (You should also link to an open issue here)

Foreign key fields are search on the attribute id for CONTAINS and EQUALS operators

**What is the new behavior?**

The operators CONTAINS and EQUALS search on the foreign key infos the criteria.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

http://192.168.31.210:8080/view/Master/job/Master_03_TMDMEE_FOR-JIRA_Slave2/62/

### CODE REVIEW

File: org.talend.mdm.query/src/com/amalto/core/query/user/TQLPredicateToMDMPredicate.java [hunk 1]
```diff
@@ -163,7 +193,9 @@ public Condition visit(FieldBetweenExpression fieldBetweenExpression) {
     @Override
     public Condition visit(FieldContainsExpression fieldContainsExpression) {
         fieldContainsExpression.getField().accept(this);
-        return new Compare(popCurrentExpression(), Predicate.CONTAINS, new StringConstant(fieldContainsExpression.getValue()));
+        return typedValues.size() > 1 ? mergeContains(Predicate.OR, fieldContainsExpression.getValue())
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.query/src/com/amalto/core/query/user/TQLPredicateToMDMPredicate.java]: To check the size greater than 1 or 0?

File: org.talend.mdm.query/src/com/amalto/core/query/user/TQLPredicateToMDMPredicate.java [hunk 2]
```diff
@@ -88,7 +108,7 @@ public Condition visit(ComparisonOperator comparisonOperator) {
 
         switch (comparisonOperator.getOperator()) {
             case EQ:
-                return new Compare(left, Predicate.EQUALS, right);
+                return typedValues.size() > 0 ? mergeEquals(Predicate.OR, left, right) : new Compare(left, Predicate.EQUALS, right);
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.query/src/com/amalto/core/query/user/TQLPredicateToMDMPredicate.java]: Here to mergeEquals() with parameter `Predicate.OR`, is this expected?

Comment (YohannTalend) [reply org.talend.mdm.query/src/com/amalto/core/query/user/TQLPredicateToMDMPredicate.java]: Yes, to check if one of foreign keys infos contains the value search by the client

File: org.talend.mdm.query/src/com/amalto/core/query/user/TQLPredicateToMDMPredicate.java (hunk 1 above)
Comment (YohannTalend) [reply org.talend.mdm.query/src/com/amalto/core/query/user/TQLPredicateToMDMPredicate.java]: Than 1 yes, if the value is 1, we just do a simple Compare and if more than 1, a condition OR contains the Compare objects

**Pull request merged**